modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/resource/v1_0/SitePageResourceImpl.java

```diff
@@ -94,6 +108,93 @@ public void deleteSiteSitePage(
 				contextUser.getUserId()));
 	}
 
+	public ExportImportDescriptor getExportImportDescriptor() {
+		return new ExportImportDescriptor() {
+
+			@Override
+			public String getItemClassName() {
+				return Layout.class.getName();
+			}
+
+			@Override
+			public Map<String, Serializable> getParameters(
+				PortletDataContext portletDataContext) {
+
+				return HashMapBuilder.<String, Serializable>put(
+					"filter",
+					() -> {
+						if ((portletDataContext.getLayoutIds() == null) ||
+							(portletDataContext.getLayoutIds().length == 0) ||
+							((portletDataContext.getLayoutIds().length == 1) &&
+							 (portletDataContext.getLayoutIds()[0] == 0))) {
+
+							return null;
+						}
+
+						Set<String> layoutExternalReferenceCodes =
+							new HashSet<>();
+
+						for (long layoutId :
+								portletDataContext.getLayoutIds()) {
+
+							Layout layout = null;
+
+							try {
+								layout = _layoutService.fetchLayout(
+									portletDataContext.getScopeGroupId(),
+									portletDataContext.isPrivateLayout(),
+									layoutId);
+							}
+							catch (PortalException portalException) {
+								if (_log.isWarnEnabled()) {
+									_log.warn(portalException);
+								}
+							}
+
+							if (layout != null) {
+								layoutExternalReferenceCodes.add(
+									layout.getExternalReferenceCode());
+							}
+						}
+
+						StringBundler sb = new StringBundler(3);
+
+						sb.append("externalReferenceCode in ('");
+
+						sb.append(
+							ListUtil.toString(
+								ListUtil.fromCollection(
+									layoutExternalReferenceCodes),
+								StringPool.BLANK, "', '"));
+
+						sb.append("')");
+
+						return sb.toString();
+					}
+				).put(
+					"siteExternalReferenceCode",
```

Comment:
Should this be a global thing? or is only applicable to pages?

Reply:
The pages API only worked this way, this might have been fixed since then.

Reply:
I agree, this could be a "global thing" when the group is not the company group